src/consensus/aft/raft.h

```diff
@@ -615,11 +656,14 @@ namespace aft
         }
       }
 
-      uint32_t offset = get_bft_offset(conf);
-      configurations.push_back({idx, std::move(conf), offset, 0});
+      if (!same_node_ids(conf, configurations.back().nodes))
+      {
+        uint32_t offset = get_bft_offset(conf);
```

Comment:
Isn't this path also used in CFT, when 2tx reconfiguration is enabled?

Reply:
Yes, except when 1tx reconfiguration is used `conf` is never the same twice. With 2-tx reconfig this happens when the only change to a configuration is a node transitioning from retiring to retired (because of the update to the nodes table).

Reply:
As per `hooks.h`, I thought that retiring nodes were marked as `std::nullopt` in the configuration, whereas retired ones were explicitly passed in the `retirees` list. 

In any case, I'm surprised this calls into BFT code (`get_bft_offset`) even though this is on the CFT path.

Reply:
The offset is there to allow BFT to run with non-consecutive node IDs; after a view-change, the next primary is the one at view-change-number + offset in the list of nodes. This has nothing to do with this PR though.

Reply:
Ok, as discussed, I think we want to remove this, but you are absolutely right that it's nothing to do with this change.